components/AudioVisualizer.tsx

```diff
@@ -24,6 +35,9 @@ export default function AudioVisualizer () {
           <IconButton aria-label='playButton' id='playPause' className='fi-rr-play' icon={<AiOutlineRight/>} w='3.5rem' h='3.5rem' borderRadius='50%'/>
           <div id='waveform'/>
         </Grid>
+        {posts.map((post:PostData) => {
```

Comment:
Can the post data be handled by a separate post component in case we need to re-use the audio visualizer somewhere?

Reply:
Not sure, because for multiple posts on one profile page I would have to dynamically render each post like this and if I wanted to render the audio visualizer I would need to pass in the post's song name to get multiple visualizers to show up

Reply:
Proposed solution: I create another component without the posts and add the plugin for microphone access, assuming you want to re-use this component for the recording modal